lib/torrent.js

```diff
@@ -638,24 +639,71 @@ export default class Torrent extends EventEmitter {
     })
   }
 
-  _verifyPieces (cb) {
-    parallelLimit(this.pieces.map((piece, index) => cb => {
+  _verifyPiece (index, cb) {
+    if (this.destroyed) return cb(new Error('torrent is destroyed'))
+
+    const getOpts = {}
+    // Specify length for the last piece in case it is zero-padded
+    if (index === this.pieces.length - 1) {
+      getOpts.length = this.lastPieceLength
+    }
+    this.store.get(index, getOpts, async (err, buf) => {
       if (this.destroyed) return cb(new Error('torrent is destroyed'))
 
-      const getOpts = {}
-      // Specify length for the last piece in case it is zero-padded
-      if (index === this.pieces.length - 1) {
-        getOpts.length = this.lastPieceLength
-      }
-      this.store.get(index, getOpts, async (err, buf) => {
-        if (this.destroyed) return cb(new Error('torrent is destroyed'))
+      if (err) return queueMicrotask(() => cb(null, false)) // ignore error
+
+      const hex = await hash(buf, 'hex')
+      if (this.destroyed) return cb(new Error('torrent is destroyed'))
 
-        if (err) return queueMicrotask(() => cb(null)) // ignore error
+      cb(null, hex === this._hashes[index])
+    })
+  }
 
-        const hex = await hash(buf, 'hex')
-        if (this.destroyed) return cb(new Error('torrent is destroyed'))
+  _verifyPiecesUsingBitfield (cb) {
+    const piecesToCheck = new Set()
+    this.bitfield.buffer = new Uint8Array(this._startupBitfield)
+    this.files.forEach(file => {
+      let checkFile = true
+      for (let i = file._startPiece; i <= file._endPiece; ++i) {
+        if (this.bitfield.get(i)) {
+          if (checkFile) {
+            piecesToCheck.add(i)
+            checkFile = false
+          }
+          this._debug('piece verified %s', i)
+          this._markVerified(i)
+        } else {
+          this._markUnverified(i)
```

Comment:
marking it as unverified isnt necessary? you only need a `this.bitfield.set(i, false)` rest can be skipped